src/lua/bindings/script.cpp

```diff
@@ -151,15 +151,26 @@ namespace lua::script
 		module->m_registered_scripts.push_back(std::move(lua_script));
 	}
 
+	// Lua API: function
+	// Table: script
+	// Name: execute_as_script
+	// Param: script_name: string: target script thread.
+	// Param: func: function: function that will be executed once in the script thread.
+	static void execute_as_script(const std::string& script_name, sol::function func)
```

Comment:
to keep it consistent with the rest of the bindings use `sol::protected_function` for the `func` param